runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -111,6 +112,8 @@
   /** A lock which has to be acquired when concurrently accessing state and timers. */
   private final ReentrantLock stateBackendLock;
 
+  private final List<InternalTimer<?, TimerInternals.TimerData>> deferredTimers = new ArrayList<>();
```

Comment:
I'd use `Dequeue` (`ArrayDequeue`) instead of a list.

Reply:
I think what is unfortunate that the only deferred timers are the cleanup timers. So this is kind of an indirection.

Reply:
Changed to use `ArrayDeque`